src/sdk/pynni/nni/compression/torch/pruning/one_shot.py

```diff
@@ -0,0 +1,149 @@
+
+            # masker.calc_mask returns None means calc_mask is not calculated sucessfully, can try later
+            if masks is not None:
+                wrapper.if_calculated = True
+            return masks
+        else:
+            return None
+
+class LevelPruner(OneshotPruner):
+    def __init__(self, model, config_list, optimizer=None):
+        super().__init__(model, config_list, pruning_algorithm='level', optimizer=optimizer)
+
+class SlimPruner(OneshotPruner):
+    def __init__(self, model, config_list, optimizer=None):
+        super().__init__(model, config_list, pruning_algorithm='slim', optimizer=optimizer)
+
+    def validate_config(self, model, config_list):
+        schema = CompressorSchema([{
+            'sparsity': And(float, lambda n: 0 < n < 1),
+            'op_types': ['BatchNorm2d'],
+            Optional('op_names'): [str]
+        }], model, logger)
+
+        schema.validate(config_list)
+
+        if len(config_list) > 1:
+            logger.warning('Slim pruner only supports 1 configuration')
+
+class _StructuredFilterPruner(OneshotPruner):
+    def __init__(self, model, config_list, pruning_algorithm, optimizer=None, **algo_kwargs):
+        super().__init__(model, config_list, pruning_algorithm=pruning_algorithm, optimizer=optimizer, **algo_kwargs)
+
+    def validate_config(self, model, config_list):
+        schema = CompressorSchema([{
+            'sparsity': And(float, lambda n: 0 < n < 1),
+            'op_types': ['Conv2d'],
+            Optional('op_names'): [str]
+        }], model, logger)
+
+        schema.validate(config_list)
+
+class L1FilterPruner(_StructuredFilterPruner):
+    def __init__(self, model, config_list, optimizer=None):
+        super().__init__(model, config_list, pruning_algorithm='l1', optimizer=optimizer)
+
+class L2FilterPruner(_StructuredFilterPruner):
+    def __init__(self, model, config_list, optimizer=None):
+        super().__init__(model, config_list, pruning_algorithm='l2', optimizer=optimizer)
+
+class FPGMPruner(_StructuredFilterPruner):
+    def __init__(self, model, config_list, optimizer=None):
+        super().__init__(model, config_list, pruning_algorithm='fpgm', optimizer=optimizer)
+
+class TaylorFOWeightFilterPruner(_StructuredFilterPruner):
+    def __init__(self, model, config_list, optimizer=None, statistics_batch_num=1):
+        super().__init__(model, config_list, pruning_algorithm='taylorfo', optimizer=optimizer, statistics_batch_num=statistics_batch_num)
+
+class ActivationRankFilterPruner(_StructuredFilterPruner):
+    def __init__(self, model, config_list, pruning_algorithm, optimizer=None, activation='relu', statistics_batch_num=1):
+        super().__init__(model, config_list, pruning_algorithm=pruning_algorithm, \
+            optimizer=optimizer, activation=activation, statistics_batch_num=statistics_batch_num)
+
+class ActivationAPoZRankFilterPruner(ActivationRankFilterPruner):
```

Comment:
why `ActivationAPoZRankFilterPruner` inherits `ActivationRankFilterPruner` rather than directly inherit `_StructuredFilterPruner`?

Reply:
Thanks, forgot to delete it.  Fixed.